### PR TITLE
Inserter: Set initial active tab ID during render

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR updates the `activeTabId` logic in the `CategoryTabs` component and sets the state during render instead of using the effect.

Initially, I planned to fix the effect dependencies (`setActiveId` can be omitted, and we could also remove `activeTabId`), but I realized we could use a different pattern.

## Why?
React will re-render `CategoryTabs` immediately and not render its children or update the DOM, avoiding rendering a tab list with an undefined active tab ID.

Details:

* https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
* https://blog.isquaredsoftware.com/2020/05/blogged-answers-a-mostly-complete-guide-to-react-rendering-behavior/#setting-state-while-rendering

## Testing Instructions
1. Open a post or page.
2. Open Inserter and switch to Patterns or Media.
3. Move focus on the tab list. Usually, two <kbd>Tab</kbd> presses when opening the pattern tab.
4. Confirm that focus correctly moves into the tab list. The first tab should be focused despite not being selected.

### Testing Instructions for Keyboard
Same.

## Screencast

https://github.com/user-attachments/assets/9cb99b34-63e1-4902-a56b-9193b6348074

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
Code change makes sense, however, I'm unable to make this work - both in `trunk` and this PR. 

Am I missing something? I expected "All" will get focused when I click on Patterns, but I didn't see it happening (I also tried navigating with tabbing).


https://github.com/user-attachments/assets/3aa8e77e-d597-429c-be3e-67a265045d80

**Comment by Mamaduka** [comment]
@tyxla, sorry for the confusion. When navigating the category list after opening the Patterns, the first category should receive focus. It usually takes two Tab presses.

I've updated the description and added the screencast.

**Review by tyxla — APPROVED** [review]
Works as expected 👍 

Thanks @Mamaduka 🚀

**Comment by Mamaduka** [comment]
Thanks for testing and review, @tyxla 🙇

File: packages/block-editor/src/components/inserter/category-tabs/index.js [hunk 1]
```diff
@@ -35,14 +35,13 @@ function CategoryTabs( {
...
-		if ( selectedTabId === null && ! activeTabId && firstTabId ) {
-			setActiveId( firstTabId );
-		}
-	}, [ selectedTabId, activeTabId, firstTabId, setActiveId ] );
+
+	// If there is no active tab, make the first tab the active tab, so that
+	// when focus is moved to the tablist, the first tab will be focused
+	// despite not being selected
+	if ( selectedTabId === null && ! activeTabId && firstTabId ) {
+		setActiveId( firstTabId );
```
Comment (ellatrix) [inline packages/block-editor/src/components/inserter/category-tabs/index.js:43]: We shouldn't set state during render?

Comment (tyxla) [reply packages/block-editor/src/components/inserter/category-tabs/index.js:43]: @ellatrix see https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes for the pattern followed here.

Comment (Mamaduka) [reply packages/block-editor/src/components/inserter/category-tabs/index.js:43]: You can if it's called conditionally; it prevents extra render.

Refs:

* https://react.dev/reference/react/useState#storing-information-from-previous-renders
* https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
* https://blog.isquaredsoftware.com/2020/05/blogged-answers-a-mostly-complete-guide-to-react-rendering-behavior/#setting-state-while-rendering

**Pull request merged**